vault/mount.go

```diff
@@ -303,6 +315,9 @@ func (c *Core) removeMountEntry(path string) error {
 
 // taintMountEntry is used to mark an entry in the mount table as tainted
 func (c *Core) taintMountEntry(path string) error {
+	c.mountsLock.Lock()
```

Comment:
This is actually being used to protect two things: the `c.mounts.SetTaint()` call and the `c.persistMounts()` call. The `mountsLock` is protecting `c.mounts` so any operation modifying it needs protection. So it needs to stay where it is.

`Tainted()` isn't actually used; I wrote it and then solved things a different way. I figured I'd leave it in but it's no good having code that is never used, so I'll just remove it for now.